File: Backend/src/main/java/com/example/backend/model/Order.java
package com.example.backend.model;

import com.example.backend.model.user.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity(name = "orders")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Date date;
    private Double totalPrice;
    private String status;

    //@JsonManagedReference
    @ManyToOne
    private User user;

    @ManyToMany(fetch = FetchType.EAGER , cascade = CascadeType.MERGE)
    //@JsonManagedReference
    @JoinTable(name = "orders_products",
            joinColumns = {@JoinColumn(name = "order_id")},
            inverseJoinColumns = {@JoinColumn(name = "product_id")})
    private List<Product> products = new ArrayList<>();
}

File: Frontend/src/app/models/order.ts
import * as moment from 'moment';
import { Product } from './product';
import { User } from './user';

export class Order {
  id: number;
  date: moment.Moment;
  totalPrice: number;
  status: string;
  user: User;
  products: Product[];

  constructor(res: Order) {
    this.id = res.id;
    this.date = moment.utc(res.date);
    this.totalPrice = res.totalPrice;
    this.status = res.status;
    this.user = res.user;
    this.products = res.products;
  }
}
